packages/angular_devkit/build_angular/src/angular-cli-files/utilities/stats.ts

```diff
@@ -105,17 +114,45 @@ export function statsWarningsToString(json: any, statsConfig: any): string {
     );
   }
 
-  return rs('\n' + warnings
-    .map((warning: any) => `${warning}`)
-    .filter(ERRONEOUS_WARNINGS_FILTER)
-    .map((warning: string) => y(`WARNING in ${warning}`))
-    .join('\n\n'));
+  let output = '';
+  for (const warning of warnings as (string | WebpackDiagnostic)[]) {
+    if (typeof warning === 'string') {
+      if (!ERRONEOUS_WARNINGS_FILTER(warning)) {
+        continue;
+      }
+      output += yb(`WARNING in ${warning}\n\n`);
+    } else {
+      if (!ERRONEOUS_WARNINGS_FILTER(warning.message)) {
+          continue;
+      }
+      const file = warning.file || warning.moduleName;
+      if (file) {
+        output += c(file);
+        if (warning.loc) {
+          output += ':' + yb(warning.loc);
+        }
+        output += ' - ';
+      }
+      if (!/^warning/i.test(warning.message)) {
+        output += y('warning: ');
```

Comment:
Nit: maybe start with a capital letter?
```suggestion
        output += y('Warning: ');
```